library/ui/src/main/java/com/google/android/exoplayer2/ui/SubtitlePainter.java

```diff
@@ -245,7 +248,14 @@ private void setupTextLayout() {
     if (applyEmbeddedFontSizes && applyEmbeddedStyles) {
       cueText = this.cueText;
     } else if (!applyEmbeddedStyles) {
-      cueText = this.cueText.toString(); // Equivalent to erasing all spans.
+      if (Color.alpha(backgroundColor) > 0) {
```

Comment:
As discussed, for this change I think we should just add the background wherever the cue hasn't provided one (which is approximately what happened before, with the exception of where the cue provided one that was semi-transparent). Let's discuss styling priorities etc separately. I'm fairly sure you can do this by adding the span at L262 and setting the flags to Spanned.SPAN_PRIORITY.